providers/efa/verbs.c

```diff
@@ -940,7 +957,10 @@ struct ibv_qp *efadv_create_qp_ex(struct ibv_context *ibvctx,
 
 	if (attr_ex->qp_type != IBV_QPT_DRIVER ||
 	    !vext_field_avail(struct efadv_qp_init_attr,
-			      driver_qp_type, inlen)) {
+			      driver_qp_type, inlen) ||
+	    efa_attr->comp_mask ||
+	    !is_reserved_cleared(efa_attr->reserved) ||
```

Comment:
If a reserved is switched to an actual value then we generally expect a comp mask bit for it

Reply:
Agree, but a non-zero value is most likely an application bug and I prefer we catch it if possible.